src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformRepositoriesHelper.kt

```diff
@@ -142,33 +146,156 @@ class IntelliJPlatformRepositoriesHelper(
+            )
+
+            // For performance reasons make it exclusive.
+            // https://docs.gradle.org/current/userguide/declaring_repositories_adv.html#declaring_content_exclusively_found_in_one_repository
+            repositories.exclusiveContent {
+                forRepository {
+                    repository
+                }
+
+                filter {
+                    for (exclusiveGroup in exclusiveGroups) {
+                        includeGroup(exclusiveGroup)
+                        // Could be improved some day to the next, but today it fails on Gradle 8.2, works on 8.10.2
+                        //includeGroupAndSubgroups(exclusiveGroup)
+                    }
+                }
+            }
+
+            repository.apply {
+                action()
+            }
+
+            return repository
+        }
 
         /**
-         * Because artifact paths always start with `/` (see [toPublication] for details),
-         * on Windows, we have to guess to which drive letter the artifact path belongs to.
-         * To do so, we add all drive letters (`a:/[artifact]`, `b:/[artifact]`, `c:/[artifact]`, ...) to the stack,
-         * starting with `c` for the sake of micro-optimization.
+         * @see org.jetbrains.intellij.platform.gradle.models.IvyModule.toIvyArtifact
+         * @see org.jetbrains.intellij.platform.gradle.models.IvyModule.toBundledIvyArtifactsRelativeTo
+         * @see org.jetbrains.intellij.platform.gradle.models.IvyModule.toLocalPluginIvyArtifacts
+         * @see org.jetbrains.intellij.platform.gradle.extensions.IntelliJPlatformDependenciesHelper.registerIntellijPlatformIvyRepo
          */
-        if (OperatingSystem.current().isWindows) {
-            (('c'..'z') + 'a' + 'b').forEach { artifactPattern("$it:/[artifact]") }
-        }
-    }.apply {
-        content {
-            includeGroup(Configurations.Dependencies.BUNDLED_MODULE_GROUP)
-            includeGroup(Configurations.Dependencies.BUNDLED_PLUGIN_GROUP)
-            includeGroup(Configurations.Dependencies.LOCAL_IDE_GROUP)
-            includeGroup(Configurations.Dependencies.LOCAL_PLUGIN_GROUP)
-            includeGroup(Configurations.Dependencies.LOCAL_JETBRAINS_RUNTIME_GROUP)
+        internal fun createIvyArtifactRepository(
+            repositoryName: String,
+            repositories: RepositoryHandler,
+            providers: ProviderFactory,
+            ivyLocationPath: Path,
+            artifactLocationPath: Path?
+        ) = repositories.ivy {
+            name = repositoryName
+
+            // The contract is that we are working with absolute normalized paths here.
+            val absNormIvyLocationPath = ivyLocationPath.absolute().normalize()
+            val absNormArtifactLocationPath = artifactLocationPath?.absolute()?.normalize()
+
+            // Location of Ivy files generated for the current project.
+            val ivyPath = providers.localPlatformArtifactsPath(absNormIvyLocationPath)
```

Comment:
`localPlatformArtifactsPath` does not make much sense with the dynamic repository. Should I apply it only for repositories which are static?

Reply:
I did that. Now the dynamic repo stores Ivy.xml files into "localPlatformArtifacts" inside `platformPath` (not customisable). If a `local("/path/to/ide")` is used, the customizable location is used from `providers.localPlatformArtifactsPath(absNormIvyLocationPath)`